internal/handshake/session_ticket.go

```diff
@@ -37,11 +40,13 @@ func (t *sessionTicket) Unmarshal(b []byte) error {
 	if err != nil {
 		return errors.New("failed to read RTT")
 	}
-	var tp wire.TransportParameters
-	if err := tp.UnmarshalFromSessionTicket(r); err != nil {
-		return fmt.Errorf("unmarshaling transport parameters from session ticket failed: %s", err.Error())
+	if using0RTT {
+		var tp wire.TransportParameters
+		if err := tp.UnmarshalFromSessionTicket(r); err != nil {
+			return fmt.Errorf("unmarshaling transport parameters from session ticket failed: %s", err.Error())
+		}
+		t.Parameters = &tp
 	}
-	t.Parameters = &tp
```

Comment:
Should we add a check here that the entire ticket has been consumed (`r.Len() == 0`)?

Reply:
I think maybe it is only needed for unmarshaling non-0-RTT session ticket.  A 0-RTT session ticket is end with transport parameters and unmarshaling transport parameters from session ticket always consumes all bytes of the session ticket(extra bytes would either be ignored as unknwon parameter IDs or cause an EOF error).